File: backend/OlivePlatform/OlivePlatform.Application/Features/Harvests/Responses/HarvestForListResponse.cs
namespace OlivePlatform.Application.Features.Harvests.Responses;

public class HarvestForListResponse
{
    public int Total { get; set; }
    public int Id { get; set; }

    public string Reference { get; set; } = null!;

    public int PlotId { get; set; }

    public int PlotReference { get; set; }

    public string PlotName { get; set; } = null!;

    public DateOnly HarvestDate { get; set; }

    public decimal QuantityKg { get; set; }

    public string? Notes { get; set; }

    public int Status { get; set; }

    public int HarvestedTrees { get; set; }

    public int VarietyId { get; set; }

    public int PlannedTrees { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? StartTime { get; set; }

    public DateTime? EndTime { get; set; }

    public DateTime? UpdatedAt { get; set; }
}

File: frontend/olive-platform/src/features/harvests/data/responses/HarvestForListResponse.ts
import type { ProductionStatus } from "../../../production/domain/entities/ProductionStatus";

export type HarvestForListResponse = {
    id: number;
    reference: string;

    plotId: number;

    plotName: string;

    harvestDate: string;

    startTime: string | null;

    endTime: string | null;

    quantityKg: number;

    qualityGrade: string | null;

    createdAt: string;

    harvestedTrees: number;

    plannedTrees: number;

    varietyId: number;

    status: number;

    notes: string | null;

    updatedAt: string;

    pressing: ProductionStatus | null;

    analysis: ProductionStatus | null;
    
    canBePressed: boolean;
};
